lux/core/sqltable.py

```diff
@@ -0,0 +1,192 @@
+from lux.vis.VisList import VisList
+from lux.history.history import History
+from lux.utils.date_utils import is_datetime_series
+from lux.utils.message import Message
+from lux.utils.utils import check_import_lux_widget
+from typing import Dict, Union, List, Callable
+
+# from lux.executor.Executor import *
+import warnings
+import traceback
+import lux
+
+
+class LuxSQLTable(lux.LuxDataFrame):
+    """
+    A subclass of pd.DataFrame that supports all dataframe operations while housing other variables and functions for generating visual recommendations.
+    """
+
+    # MUST register here for new properties!!
+    _metadata = [
+        "_intent",
+        "_inferred_intent",
+        "_data_type",
+        "unique_values",
+        "cardinality",
+        "_rec_info",
+        "_min_max",
+        "_current_vis",
+        "_widget",
+        "_recommendation",
+        "_prev",
+        "_history",
+        "_saved_export",
+        "_sampled",
+        "_toggle_pandas_display",
+        "_message",
+        "_pandas_only",
+        "pre_aggregated",
+        "_type_override",
+    ]
+
+    def __init__(self, *args, table_name="", **kw):
+        self._history = History()
+        self._intent = []
+        self._inferred_intent = []
+        self._recommendation = {}
+        self._saved_export = None
+        self._current_vis = []
+        self._prev = None
+        self._widget = None
+        super(LuxSQLTable, self).__init__(*args, **kw)
+        from lux.executor.SQLExecutor import SQLExecutor
+
+        lux.config.executor = SQLExecutor()
+
+        self._sampled = None
+        self._toggle_pandas_display = True
+        self._message = Message()
+        self._pandas_only = False
+        # Metadata
+        self._data_type = {}
+        self.unique_values = None
+        self.cardinality = None
+        self._min_max = None
+        self.pre_aggregated = None
+        self._type_override = {}
+
+        if table_name != "":
+            self.set_SQL_table(table_name)
+        warnings.formatwarning = lux.warning_format
+
+    def set_SQL_table(self, t_name):
+        # function that ties the Lux Dataframe to a SQL database table
+        if self.table_name != "":
+            warnings.warn(
+                f"\nThis dataframe is already tied to a database table. Please create a new Lux dataframe and connect it to your table '{t_name}'.",
+                stacklevel=2,
+            )
+        else:
+            self.table_name = t_name
+        import psycopg2
+
+        try:
+            lux.config.executor.compute_dataset_metadata(self)
+        except Exception as error:
+            error_str = str(error)
+            if f'relation "{t_name}" does not exist' in error_str:
+                warnings.warn(
+                    f"\nThe table '{t_name}' does not exist in your database./",
+                    stacklevel=2,
+                )
+
+    def _repr_html_(self):
+        from IPython.display import display
```

Comment:
I assume this will have to change with @westernguy2 's changes. Is there some way you can again re-use `super._repr_html_()` or what it becomes as per #308 ?